src/common/datas/ChanceTable.ts

```diff
@@ -0,0 +1,65 @@
+export interface TableEntry<T> {
+  value: T;
+  rarity: number;
+}
+
+export class ChanceTable<T> {
+  private entries: TableEntry<T>[];
+
+  /**
+   * A chance table for getting entries based on rarity.
+   * @param entries Array of table entries.
+   */
+  constructor(entries: TableEntry<T>[]) {
+    this.entries = entries;
+  }
+
+  /**
+   * Returns a randomly selected entry from LootTable
```

Comment:
Update JSDoc to match class name. Use @returns for descibing the returned value in more detail.